### ISSUE TITLE
missing_asserts_for_indexing + slice.windows(..)

### ISSUE THREAD

**Issue by elbaro** [rust-lang/rust-clippy#17159]
### Summary

`[1, 2, 3, 4].windows(2)` always emits a slice of length 2, but `clippy::missing-asserts-for-indexing` doesn't know this.

### Lint Name

clippy::missing-asserts-for-indexing

### Reproducer

```rust
fn main() {
    for pair in [1, 2, 3, 4].windows(2) {
        let sum = pair[0] + pair[1];
        println!("{sum}");
    }
}

```

```sh
  warning: indexing into a slice multiple times without an `assert`
   --> demo.rs:3:19
    |
  3 |         let sum = pair[0] + pair[1];
    |                   ^^^^^^^   ^^^^^^^
    |
    = help: consider asserting the length before indexing: `assert!(pair.len() > 1);`
    = note: asserting the length before indexing will elide bounds checks
    = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.96.0/index.html#missing_asserts_for_indexing
    = note: requested on the command line with `-W clippy::missing-asserts-for-indexing`

  warning: 1 warning emitted
```

### Version

```text
rustc 1.96.0 (ac68faa20 2026-05-25)
binary: rustc
commit-hash: ac68faa20c58cbccd01ee7208bf3b6e93a7d7f96
commit-date: 2026-05-25
host: x86_64-unknown-linux-gnu
release: 1.96.0
LLVM version: 22.1.2
```

**Comment by elbaro** [comment]
Actually replacing `.windows(2)` with `.array_windows(2)` per new lint in #6580 is a better solution
